interfaces/builtin/docker_test.go

```diff
@@ -0,0 +1,116 @@
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+)
+
+type DockerInterfaceSuite struct {
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&DockerInterfaceSuite{
+	iface: &builtin.DockerInterface{},
+	slot: &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "docker"},
+			Name:      "docker-daemon",
+			Interface: "docker",
+		},
+	},
+	plug: &interfaces.Plug{
+		PlugInfo: &snap.PlugInfo{
+			Snap:      &snap.Info{SuggestedName: "docker"},
+			Name:      "docker-client",
+			Interface: "docker",
+		},
+	},
+})
+
+func (s *DockerInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "docker")
+}
+
+// TODO more tests regarding the contents of Docker's security profiles (especially AppArmor)?
+
+func (s *DockerInterfaceSuite) TestUnusedSecuritySystems(c *C) {
+	for _, system := range [...]interfaces.SecuritySystem{
+		interfaces.SecurityAppArmor,
+		interfaces.SecurityDBus,
+		interfaces.SecurityMount,
+		interfaces.SecuritySecComp,
+		interfaces.SecurityUDev,
+	} {
+		snippet, err := s.iface.PermanentPlugSnippet(s.plug, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, IsNil)
+		snippet, err = s.iface.ConnectedSlotSnippet(s.plug, s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, IsNil)
+	}
+	for _, system := range [...]interfaces.SecuritySystem{
+		interfaces.SecurityDBus,
+		interfaces.SecurityMount,
+		interfaces.SecurityUDev,
+	} {
+		snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, IsNil)
+		snippet, err = s.iface.PermanentSlotSnippet(s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, IsNil)
+	}
+}
+
+func (s *DockerInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	for _, system := range [...]interfaces.SecuritySystem{
+		interfaces.SecurityAppArmor,
+		interfaces.SecuritySecComp,
+	} {
+		snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, Not(IsNil))
+		snippet, err = s.iface.PermanentSlotSnippet(s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, Not(IsNil))
+	}
+}
+
+func (s *DockerInterfaceSuite) TestUnexpectedSecuritySystems(c *C) {
+	snippet, err := s.iface.PermanentPlugSnippet(s.plug, "foo")
+	c.Assert(err, Equals, interfaces.ErrUnknownSecurity)
+	c.Assert(snippet, IsNil)
+	snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, "foo")
+	c.Assert(err, Equals, interfaces.ErrUnknownSecurity)
+	c.Assert(snippet, IsNil)
+	snippet, err = s.iface.PermanentSlotSnippet(s.slot, "foo")
+	c.Assert(err, Equals, interfaces.ErrUnknownSecurity)
+	c.Assert(snippet, IsNil)
+	snippet, err = s.iface.ConnectedSlotSnippet(s.plug, s.slot, "foo")
+	c.Assert(err, Equals, interfaces.ErrUnknownSecurity)
+	c.Assert(snippet, IsNil)
+}
```

Comment:
Please also add a test for TestConnectedPlugSnippet() and search for the docker socket rule for apparmor and bind for seccomp. You can see an example of this sort of thing in mpris_test.go (but there are others).

Please do the same for TestPermanentSlotSnippet() with an appropriate string search for both apparmor and seccomp.